---
id: faq
title: FAQ
sidebar_label: FAQ
sidebar_position: 6
---

## General questions



<details>
<summary>Whom is Fused for?</summary>

Fused is built for individuals & teams wanting to get things done quickly with data. 

From building quick prototypes with AI by dragging & dropping a dataset to start working with, to making professional dashboards with large dashboards. 

{/* TODO: Link to Analytics blog once its up */}
</details>


<details>
<summary>Which authentication methods do you support?</summary>

Fused currently uses Auth0 to support authentication via Google and GitHub.
</details>


## Troubleshooting

### Status page

Access our [status page](https://fused.instatus.com/) at any time to check on Workbench & API status

### Common Errors

<details>
<summary>Error: `Access is not configured for you in the Fused Workbench. Please refresh the page if you think this is an error, or get in touch if you require further help. Cause: Realtime instance not configured.`</summary>

This error occurs when you try to run a UDF with an account associated with a workspace environment that does not have a realtime instance configured. This means that there are no worker nodes available to run the UDF. To resolve this issue, please get in touch with the Fused team team to ensure your account is associated with an environment with a realtime instance.

When Troubleshooting this error, it may help to navigate to your account's User Profile page to determine if the account is associated with an environment and realtime instance, as shown here.

![user settings](/img/user_settings.png)

</details>



<details>
<summary>Error: `No such file or directory: '/mnt/cache/'`</summary>

This error occurs when a UDF attempts to access the [`/mnt/cache` disk](/core-concepts/content-management/file-system/#mntcache-disk) when it is not available for the environment. To resolve this issue, please contact the Fused team to ensure that the cache directory is available for your account.
</details>

<details>
<summary>Error: `No space left on the device: '/tmp/'`</summary>

This error occurs when a UDF attempts to write more data than the `/tmp` directory of the real-time instance can handle. Realtime instances have a limited amount of space available and are ephemeral between runs. You might want to consider writing to [`/mnt/cache` disk](/core-concepts/content-management/file-system/#mntcache-disk) instead.
</details>

<details>
<summary>Error: `Quota limit: Number of running instances`</summary>

Fused batch jobs, which are initiated with [run_remote](/python-sdk/top-level-functions/#jobrun_remote), require a server quota to be enabled for your account. These include data [ingestion](/core-concepts/data-ingestion/) jobs. If you encounter this error, please contact the Fused team to request an increase in the quota allotted to your account.
</details>

<details>
<summary>Error: `Application error: a client-side exception has occurred (see browser console for more information).`</summary>

In the case that you encounter this error, please reset your browser cache and cookies. If the error persists, please contact the Fused team for further assistance.
</details>


<details>
<summary>Error: `MY_UDF... not saved because a UDF or app wit that name already exists.`</summary>

It's possible you might be editing a "local" UDF that has the same name as a Team UDF. To resolve this issue, you can either rename the UDF or, if your intention is to update the Team UDF, [Push it to GitHub](/workbench/udf-catalog/#publish-a-udf-to-a-github-repository).

</details>


<details>
<summary>Error: `Failed to create share token: {"detail":"UDF not found"}`</summary>

To resolve this error, save the UDF again and refreshing your browser window. If the problem persists, rename the UDF and try again.

</details>

<details>
<summary>Error: `ModuleNotFoundError: No module named 'geopandas' / `xarray`</summary>

This error happens when your UDF returns a `GeoDataFrame` or an `xarray` Dataset but your local env doesn't have the `geopandas` or `xarray` package installed to manipulate it.

The standalone `fused` package doesn't have `geopandas` or `xarray` or other optional dependencies installed to keep it light weight. 

This can be fixed in 2 ways:

1. When running fused locally

You can [install](/python-sdk/#python-install) `fused` with all the optional dependencies:

```bash
pip install "fused[all]"
```

2. When running a UDF in a [Fused App](/workbench/app-builder/)

Install the package using `micropip` at the top of your app:

```python
import micropip
await micropip.install(['geopandas'])
# await micropip.install(['xarray']) # if you were missing xarray
import geopandas as gpd
```

You can install multiple packages the same way:
```python
await micropip.install(['geopandas', 'xarray'])
```

</details>


<details>
<summary>Workbench Memory Usage tells me "Unknown"</summary>

We try our best to retrieve memory usage for UDFs you run in Workbench to help you monitor & optimize your UDFs.

There are a few reasons why we might not be able to retrieve memory usage:
1. **Running a shared token, for example a public UDF**: In this case we don't share memory usage as the original builder of the UDF might not want to reveal the inner workings of their UDF.
2. **Running a [cached UDF](/core-concepts/cache/)**: If you're running a UDF that was cached (rerunning the same UDF with the same parameters), no memory was used. 

</details>